File: simpleApi/src/main/java/com/lifemmo/pl/simple/api/ApiSystem.java
package com.lifemmo.pl.simple.api;

import com.lifemmo.pl.base.protocol.Packet;
import com.lifemmo.pl.base.protocol.ReplyPacket;
import com.lifemmo.pl.base.service.BaseSender;
import com.lifemmo.pl.base.service.ServerSystem;
import com.lifemmo.pl.base.service.SystemPanel;
import com.lifemmo.pl.simple.Simple;
import lombok.extern.slf4j.Slf4j;
import org.apache.commons.lang3.exception.ExceptionUtils;
import org.jetbrains.annotations.NotNull;

@Slf4j
public class ApiSystem implements ServerSystem {

    private BaseSender baseSender;
    private SystemPanel systemPanel;

    ApiSystem(SystemPanel systemPanel,BaseSender baseSender){
        this.baseSender = baseSender;
        this.systemPanel = systemPanel;
    }



    @Override
    public void onDispatch(@NotNull Packet packet) {
        log.info("{} packet received",packet.getMsgName());

        try{
            if(packet.getMsgName().equals(Simple.HelloReq.getDescriptor().getName())){
                var data = Simple.HelloReq.parseFrom(packet.buffer()).getMessage();
                this.baseSender.reply(new ReplyPacket(Simple.HelloRes.newBuilder().setMessage(data).build()));
            }
        }catch (Exception e){
            log.error(ExceptionUtils.getMessage(e));
        }

    }

    @NotNull
    @Override
    public BaseSender getBaseSender() {
        return this.baseSender;
    }

    @NotNull
    @Override
    public SystemPanel getSystemPanel() {
        return this.systemPanel;
    }


    @Override
    public void onPause() {

    }

    @Override
    public void onResume() {

    }

    @Override
    public void onStart() {

    }

    @Override
    public void onStop() {

    }


}

File: simpleApi/src/main/kotlin/org/ulalax/playhouse/simple/api/ApiSystem.kt
package org.ulalax.playhouse.simple.api

import org.apache.commons.lang3.exception.ExceptionUtils
import org.apache.logging.log4j.kotlin.logger
import org.ulalax.playhouse.communicator.message.Packet
import org.ulalax.playhouse.communicator.message.ReplyPacket
import org.ulalax.playhouse.service.Sender
import org.ulalax.playhouse.service.ServerSystem
import org.ulalax.playhouse.service.SystemPanel
import org.ulalax.playhouse.simple.Simple.*


class ApiSystem(override val systemPanel: SystemPanel,override val sender: Sender) : ServerSystem {
    private val log = logger()

    override suspend fun onDispatch(packet: Packet) {
        log.info("${packet.msgId} packet received")

        try {
            if (packet.msgId == HelloReq.getDescriptor().index) {
                val data: String = HelloReq.parseFrom(packet.data()).message
                sender.reply(ReplyPacket(HelloRes.newBuilder().setMessage(data).build()))
            }
        } catch (e: Exception) {
            log.error(ExceptionUtils.getMessage(e))
        }
    }

    override suspend fun onPause() {
        log.info("onPause")
    }

    override suspend fun onResume() {
        log.info("onResume")
    }

    override suspend fun onStart() {
        log.info("onStart")
    }

    override suspend fun onStop() {
        log.info("onStop")
    }
}